Test/FMI3/fmi3_import_variable_types_test.cpp

```diff
@@ -0,0 +1,118 @@
+    fmi3_float32_t max_ref =        FLT_MAX;
+    fmi3_float32_t nominal_ref =    1.00000000f;
+    fmi3_float32_t start_ref =      3.14159274f;
+
+    fmi3_import_float32_variable_t* var = fmi3_import_get_variable_as_float32(v);
+    REQUIRE(var != nullptr);
+
+    REQUIRE(fmi3_import_get_float32_variable_start(var) == start_ref); /* user defined */
+    REQUIRE(fmi3_import_get_float32_variable_min(var) == min_ref); /* default */
+    REQUIRE(fmi3_import_get_float32_variable_max(var) == max_ref); /* default */
+    REQUIRE(fmi3_import_get_float32_variable_nominal(var) == nominal_ref); /* default */
+}
+
+/* Parse user defined: min, max, nominal for float64 variable*/
+static void test_full_float64(fmi3_import_t* xml) {
+    fmi3_import_variable_t* v = fmi3_import_get_variable_by_name(xml, "var3");
+    REQUIRE(v != nullptr);
+    REQUIRE(fmi3_import_get_variable_vr(v) == 3);
+    
+    /* Arbitrarily chosen values that don't get truncated/rounded */
+    fmi3_float64_t min_ref =       -3500.0000000000000;
+    fmi3_float64_t max_ref =        3343492.7982000001;
+    fmi3_float64_t nominal_ref =    7.7776999499999997;
+
+    fmi3_import_float64_variable_t* var = fmi3_import_get_variable_as_float64(v);
+    REQUIRE(var != nullptr);
+
+    REQUIRE(fmi3_import_get_float64_variable_min(var) == min_ref); /* user defined */
+    REQUIRE(fmi3_import_get_float64_variable_max(var) == max_ref); /* user defined */
+    REQUIRE(fmi3_import_get_float64_variable_nominal(var) == nominal_ref); /* user defined */
+}
+
+/* Parse user defined: min, max, nominal for float32 variable*/
+static void test_full_float_32(fmi3_import_t* xml) {
+    fmi3_import_variable_t* v = fmi3_import_get_variable_by_name(xml, "var4");
+    REQUIRE(v != nullptr);
+    REQUIRE(fmi3_import_get_variable_vr(v) == 4);
+
+    /* Arbitrarily chosen values that don't get truncated/rounded */
+    fmi3_float32_t min_ref =       -3500.00000f;
+    fmi3_float32_t max_ref =        3343492.75f;
+    fmi3_float32_t nominal_ref =    7.77769995f;
+
+    fmi3_import_float32_variable_t* var = fmi3_import_get_variable_as_float32(v);;
+    REQUIRE(var != nullptr);
+
+    REQUIRE(fmi3_import_get_float32_variable_min(var) == min_ref); /* user defined */
+    REQUIRE(fmi3_import_get_float32_variable_max(var) == max_ref); /* user defined */
+    REQUIRE(fmi3_import_get_float32_variable_nominal(var) == nominal_ref); /* user defined */
+}
+
+TEST_CASE("Varibles types testing") {
+    const char* xmldir = FMI3_TEST_XML_DIR "/variable_types/float";
+
+    jm_callbacks* cb = jm_get_default_callbacks();
+    fmi_import_context_t* ctx = fmi_import_allocate_context(cb);
+    fmi3_import_t* xml = fmi3_import_parse_xml(ctx, xmldir, nullptr);
+
+    REQUIRE(ctx != nullptr);
+    REQUIRE(xml != nullptr);
+
+    test_small_float64(xml);
+    test_small_float32(xml);
+
+    test_full_float64(xml);
+    test_full_float_32(xml);
+
+    fmi_import_free_context(ctx);
+    fmi3_import_free(xml);
+}
```

Comment:
Same comments here as in the previous test file, i.e. copyright notice and inconsistent use of comment formatters.

Reply:
Fixed.